query-engine/connectors/sql-query-connector/src/database/connection.rs

```diff
@@ -14,15 +15,22 @@ use std::collections::HashMap;
 pub struct SqlConnection<C> {
     inner: C,
     connection_info: ConnectionInfo,
+    features: Vec<PreviewFeature>,
 }
 
 impl<C> SqlConnection<C>
 where
     C: QueryExt + Send + Sync + 'static,
 {
-    pub fn new(inner: C, connection_info: &ConnectionInfo) -> Self {
+    pub fn new(inner: C, connection_info: &ConnectionInfo, features: &[PreviewFeature]) -> Self {
         let connection_info = connection_info.clone();
-        Self { inner, connection_info }
+        let features = features.to_owned();
```

Comment:
If you need to clone it in the function it's better to require an owned type on the function, it communicates what happens a lot better.